scene/gui/tree.cpp

```diff
@@ -3868,6 +3868,9 @@ void Tree::gui_input(const Ref<InputEvent> &p_event) {
 				}
 
 				if (!root || (!root->get_first_child() && hide_root)) {
+					if (hide_root) {
```

Comment:
This condition should be removed.
The signal is not emitted when `hide_root` is disabled and there are no items.

Reply:
Removed the if statement so now it should trigger when there are items and no items